### PR TITLE
Weekly challenge

### PR DESCRIPTION
**Pull request by ellsclytn**
Adds content from Prismic, and metrics from Supporter. Selects active weekly challenge based on current timestamp

### CODE REVIEW

**Comment by danhenderson** [comment]
Flow issues @ellsclytn

File: source/store/weeklyChallenge/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    repository: 'cycling-cares',
+    type: 'weekly-challenge'
+  })
+)
+
+const fetchedWeeklyChallenge = (state: {}, { data: documents }: { data: Array<any> }) => ({
+  status: 'fetched',
+  data: documents.map((doc) => (
+    (deserializeDocument(doc): WeeklyChallengeContent)
+  )).find((doc) => (
```
Comment (danhenderson) [inline source/store/weeklyChallenge/index.js]: Will `find` work in IE?

File: source/store/weeklyChallenge/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  createDocumentsAction(c.NAMESPACE, {
+    repository: 'cycling-cares',
+    type: 'weekly-challenge'
+  })
+)
+
+const fetchedWeeklyChallenge = (state: {}, { data: documents }: { data: Array<any> }) => ({
+  status: 'fetched',
+  data: documents.map((doc) => (
+    (deserializeDocument(doc): WeeklyChallengeContent)
```
Comment (danhenderson) [inline source/store/weeklyChallenge/index.js]: This line my head spin a little, but I get it now. WeeklyChallengeContent is obviously a flow type.

File: source/store/weeklyMetrics/index.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+const fetchFitnessTotals = ({
+  start,
+  end
+}: {
+  start: string,
+  end: string
+}) => (
+  axios.get(`${supporterUrl}/api/v3/prerelease/metrics/campaigns/${campaignUuid}`, {params: {
+    'filter[fitness_activities]start_at': undefined,
+    'filter[fitness_activities]end_at': undefined
```
Comment (danhenderson) [inline source/store/weeklyMetrics/index.js]: Should this be using `start` and `end`?

File: source/store/weeklyChallenge/index.js (hunk 2 above)
Comment (ellsclytn) [reply source/store/weeklyChallenge/index.js]: Yeah, this is typecasting, seeing as Axios of course won't have knowledge of the shape of the object that it responds with. Was easiest to cast the type _after_ it has been through Prismic Utils deserialisation.

File: source/store/weeklyChallenge/index.js (hunk 1 above)
Comment (ellsclytn) [reply source/store/weeklyChallenge/index.js]: I hope so.

Comment (ellsclytn) [reply source/store/weeklyChallenge/index.js]: Okay, I polyfilled it.

**Pull request merged**